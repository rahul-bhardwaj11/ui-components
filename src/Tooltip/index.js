import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import AntTooltip from 'antd/lib/tooltip';
import 'antd/lib/tooltip/style/index.css';
import mixins from '../styles/mixins.js';
import ReactDOM from 'react-dom';

const MtTooltip = styled.div`
  display: inline-block;
  .ant-tooltip-inner {
    min-height: unset;
    padding: 4px 8px 6px;
    ${mixins.smallInactiveLink()};
    color: #ffffff;
    font-weight: 600;
    background-color: #000000;
  }
  .ant-tooltip {
    z-index: ${({ style = {} }) =>
      style.zIndex ? style.zIndex : mixins.zIndex.TOOLTIP};
    ${({ style = {} }) => (style.left ? `left: ${style.left} !important` : '')};
  }
`;
class Tooltip extends Component {
  static propTypes = {
    children: PropTypes.node,
    title: PropTypes.oneOfType([PropTypes.node, PropTypes.string]),
    className: PropTypes.string,
    getPopupContainer: PropTypes.func,
    style: PropTypes.object
  };

  popUpContainer = null;
  element = null;
  constructor(p) {
    super(p);
    this.element = document.createElement('div');
    this.element.style.height = 0;
    this.element.style.lineHeight = 0;
  }

  static defaultProps = {
    style: {}
  };

  componentDidMount() {
    document.body.appendChild(this.element);
  }

  componentWillUnmount() {
    document.body.removeChild(this.element);
  }

  getRef = () => {
    return this.popUpContainer;
  };

  render() {
    const { children, className, getPopupContainer, style } = this.props;
    const container = getPopupContainer && getPopupContainer();
    return (
      <React.Fragment>
        {ReactDOM.createPortal(
          <MtTooltip
            style={style}
            className={className}
            innerRef={e => e && (this.popUpContainer = e)}
          />,
          container || this.element
        )}
        <MtTooltip className={className} innerRef={this.getRef}>
          <AntTooltip
            {...this.props}
            getPopupContainer={() => {
              return this.popUpContainer;
            }}
          >
            {children}
          </AntTooltip>
        </MtTooltip>
      </React.Fragment>
    );
  }
}

export default Tooltip;
