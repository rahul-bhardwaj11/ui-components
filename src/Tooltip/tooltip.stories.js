import React from 'react';

import { storiesOf } from '@storybook/react';
import Tooltip from './index';
import { withInfo } from '@storybook/addon-info';
import { withKnobs } from '@storybook/addon-knobs';

const stories = storiesOf('Tooltip', module);
stories.addDecorator(withKnobs);

stories.add(
  'Tooltip',
  withInfo('Basic usage tooltip')(() => (
    <div>
      <Tooltip title="prompt text">
        <span>Tooltip will show when mouse enter.</span>
      </Tooltip>
    </div>
  ))
);
