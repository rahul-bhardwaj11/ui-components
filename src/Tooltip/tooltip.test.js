import React from 'react';
import renderer from 'react-test-renderer';
import Tooltip from './index';

describe('Tooltip', () => {
  it('matches default snapshot', () => {
    const tree = renderer
      .create(
        <Tooltip title="prompt text">
          <span>Tooltip will show when mouse enter.</span>
        </Tooltip>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
