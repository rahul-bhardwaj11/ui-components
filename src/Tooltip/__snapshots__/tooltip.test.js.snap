// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip matches default snapshot 1`] = `
<div
  className="sc-bdVaJa gBghzs"
>
  <span
    onBlur={undefined}
    onClick={undefined}
    onContextMenu={undefined}
    onFocus={undefined}
    onMouseDown={undefined}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onTouchStart={undefined}
  >
    Tooltip will show when mouse enter.
  </span>
</div>
`;
